apps/charge-smart-wallets-service/src/data-layer/data-layer.service.ts

```diff
@@ -23,12 +25,12 @@ export class DataLayerService {
     @Inject(walletActionString)
     private paginatedWalletActionModel: PaginateModel<WalletActionDocument>,
     private userOpFactory: UserOpFactory,
-    private tokenService: TokenService
+    private tokenService: TokenService,
+    private smartWalletsEventsService: SmartWalletsEventsService
   ) { }
 
   async recordUserOp (baseUserOp: BaseUserOp) {
     const userOp = await this.userOpFactory.createUserOp(baseUserOp)
-
     const response = this.userOpModel.create(userOp)
     this.createWalletActionFromUserOp(userOp)
```

Comment:
replace publish of walletAction to this fucntion 
and also add publish of userOp function itself